layer/nrlf/core/validators.py

```diff
@@ -268,7 +268,21 @@ def _validate_ssp_asid(self, model: DocumentReference):
                 if content.attachment.url.startswith("ssp://")
             ]
         )
-        if not ssp_content:
+        does_related_exist = getattr(model.context, "related", None)
+        does_asid_exist = False
+        if does_related_exist:
+            asid_references = [
```

Comment:
Functionally seems to work but I'm wondering if we could re-organise things a bit, to try and simplify the logic :thinking: 

Maybe something like:

1. Get SSP content
2. Get ASIDs
3. If ASIDs - validate that there's only one valid ASID
4. If SSP content but no ASID - add_error

If something like that would work, it would make the logic in this function simpler and would possibly minimise duplicated checks and such.

Just a suggestion mind, logic-wise the code is working for me.

Reply:
made some changes to simplify the logic and reduce duplication